test/performance/perf_test.lua

```diff
@@ -0,0 +1,400 @@
+    -- to test compatibility as part of unit/integration test run.
+    g.perf_mode_on = os.getenv('PERF_MODE_ON')
+
+    g.total_report = {}
+end)
+
+g.before_each(function(g)
+    helpers.truncate_space_on_cluster(g.cluster, 'customers')
+    reset_gen()
+end)
+
+local function normalize(s, n)
+    if type(s) == 'number' then
+        s = ('%.2f'):format(s)
+    end
+
+    local len = s:len()
+    if len > n then
+        return s:sub(1, n)
+    end
+
+    return (' '):rep(n - len) .. s
+end
+
+local column_name = {
+    without_stats_wrapper = 'without stats wrapper',
+    stats_disabled = 'stats disabled',
+    local_stats = 'local stats',
+    metrics_stats = 'metrics stats',
+}
+
+local function visualize_section(total_report, name, section, params)
+    local report_str = ('== %s ==\n\n'):format(name)
+
+    local headers = normalize('', params.row_header_width - 1) .. ' ||'
+
+    for _, column in ipairs(params.columns) do
+        headers = headers .. ' ' .. normalize(column, params.col_width - 2) .. ' |'
+    end
+
+    report_str = report_str .. headers .. '\n'
+    report_str = report_str .. ('='):rep(headers:len()) .. '\n'
+
+    for row_name, row in pairs(total_report) do
+        local row_str = normalize(row_name, params.row_header_width - 1) .. ' ||'
+
+        for _, column in ipairs(params.columns) do
+            local report = row[column]
+
+            if report ~= nil then
+                row_str = row_str .. ' ' .. normalize(report.str[section], params.col_width - 2) .. ' |'
+            else
+                row_str = row_str .. ' ' .. normalize('unknown', params.col_width - 2) .. ' |'
+            end
+        end
+
+        report_str = report_str .. row_str .. '\n'
+        report_str = report_str .. ('-'):rep(row_str:len()) .. '\n'
+    end
+
+    report_str = report_str .. '\n\n\n'
+
+    return report_str
+end
+
+local function visualize_report(report)
+    local params = {}
+
+    params.col_width = 2
+    for _, name in pairs(column_name) do
+        params.col_width = math.max(name:len() + 2, params.col_width)
+    end
+
+    params.row_header_width = 30
+
+    -- Set columns explicitly to preserve custom order.
+    params.columns = {
+        column_name.without_stats_wrapper,
+        column_name.stats_disabled,
+        column_name.local_stats,
+        column_name.metrics_stats,
+    }
+
+    local report_str = '\n==== PERFORMANCE REPORT ====\n\n\n'
+
+    report_str = report_str .. visualize_section(report, 'SUCCESS REQUESTS', 'success_count', params)
+    report_str = report_str .. visualize_section(report, 'SUCCESS REQUESTS PER SECOND', 'success_rps', params)
+    report_str = report_str .. visualize_section(report, 'ERRORS', 'error_count', params)
+    report_str = report_str .. visualize_section(report, 'AVERAGE CALL TIME', 'average_time', params)
+    report_str = report_str .. visualize_section(report, 'MAX CALL TIME', 'max_time', params)
```

Comment:
please add to each metric how to evaluate numbers ("lower is better", "higher is better" or something else).

Reply:
Added a comment to output record